libp2p/Session.h

```diff
@@ -158,11 +193,25 @@ class Session: public std::enable_shared_from_this<Session>
 	std::map<uint16_t, std::shared_ptr<Framing> > m_framing;
 	std::deque<bytes> m_encFrames;
 
+	static bool isFramingAllowedForVersion(unsigned _version) { return _version > 4; }
 	bool isFramingEnabled() const { return isFramingAllowedForVersion(m_info.protocolVersion); }
 	unsigned maxFrameSize() const { return 1024; }
 	std::shared_ptr<Framing> getFraming(uint16_t _protocolID);
 	void multiplexAll();
 };
 
+template <class PeerCap>
+std::shared_ptr<PeerCap> capabilityFromSession(SessionFace const& _session, u256 const& _version = PeerCap::version())
+{ 
+	try 
+	{ 
+		return std::static_pointer_cast<PeerCap>(_session.capabilities().at(std::make_pair(PeerCap::name(), _version)));
```

Comment:
This is only used in networking code, right? So we could spend the extra CPU cycles and use `dynamic_pointer_cast` for added safety. What do you think?